UnityProject/Assets/Scripts/Managers/SoundManager.cs

```diff
@@ -83,156 +103,87 @@ private void Init()
+	/// <returns>The addressable audio source with it's component loaded</returns>
+	private static async Task<AddressableAudioSource> EnsureAddressableAudioSourceFromCache(List<AddressableAudioSource> addressableAudioSources)
 	{
-		ReinitSoundPool();
-	}
+		AddressableAudioSource addressableAudioSource = addressableAudioSources.PickRandom();
+		AddressableAudioSource addressableAudioSourceFromCache = Instance.SoundsLibrary.FirstOrDefault(p => p.Path == addressableAudioSource.Path);
 
-	private void ReinitSoundPool()
-	{
-		for (int i = Instance.pooledSources.Count - 1; i > 0; i--)
+		if (addressableAudioSourceFromCache == null)
 		{
-			if (Instance.pooledSources[i] != null)
-			{
-				Destroy(Instance.pooledSources[i].gameObject);
-			}
+			Instance.SoundsLibrary.Add(addressableAudioSource);
 		}
 
-		Instance.pooledSources.Clear();
+		// Ensure it's loaded and valid
+		AudioSource audioSource;
+		GameObject gameObject = await addressableAudioSource.Load();
 
-		// Cache some pooled sources:
-		for (int i = 0; i < 20; i++)
+		if (!gameObject.TryGetComponent(out audioSource))
 		{
-			var soundObj = Instantiate(soundSpawnPrefab, transform);
-			pooledSources.Add(soundObj.GetComponent<SoundSpawn>());
+			Logger.LogError($"AddressableAudioSource in SoundManager doesn't contain an AudioSource: {addressableAudioSource.Path}", Category.Addressables);
+			return null;
 		}
+
+		return addressableAudioSource;
 	}
 
 	/// <summary>
-	/// Uses a pooled AudioSource instead of the original one.
-	/// This copies the sourceToCopy settings to a source taken from the pool
-	/// and return it.
+	/// Get a fully loaded addressableAudioSource from the loaded cache.  This ensures that everything is ready to use.
+	/// If more than one addressableAudioSource is provided, one will be picked at random.
 	/// </summary>
-	private SoundSpawn GetSourceFromPool(AudioSource sourceToCopy)
+	/// <param name="addressableAudioSources">The sound to be played.  If more than one is specified, one will be picked at random.</param>
+	/// <returns>A fully loaded and ready to use AddressableAudioSource</returns>
+	public static async Task<AddressableAudioSource> GetAddressableAudioSourceFromCache(List<AddressableAudioSource> addressableAudioSources)
 	{
-		for (int i = pooledSources.Count - 1; i > 0; i--)
-		{
-			if (pooledSources[i] != null && pooledSources[i].gameObject != null
-										 && !pooledSources[i].isPlaying)
-			{
-				pooledSources[i].isPlaying = true;
-				CopySource(pooledSources[i].audioSource, sourceToCopy);
-				return pooledSources[i];
-			}
-		}
-
-		var soundObj = Instantiate(soundSpawnPrefab, transform);
-		var source = soundObj.GetComponent<SoundSpawn>();
-		pooledSources.Add(source);
-		source.isPlaying = true;
-		CopySource(source.audioSource, sourceToCopy);
-		return source;
+		return await EnsureAddressableAudioSourceFromCache(addressableAudioSources);
```

Comment:
Codacy found an issue: [Add '.ConfigureAwait(false)' to this call to allow execution to continue in any thread.](https://app.codacy.com/gh/unitystation/unitystation/pullRequest?prid=6280689)